examples/AdvancedOperations/AddDisplayUploadAd.php

```diff
@@ -0,0 +1,243 @@
+                PHP_EOL
+            );
+            exit(1);
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $adGroupId the ad group ID to add a display upload ad to
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId
+    ) {
+        // There are several types of display upload ads. For this example, we will create
+        // an HTML5 upload ad, which requires a media bundle.
+        // The DisplayUploadProductType field lists the available display upload types:
+        // https://developers.google.com/google-ads/api/reference/rpc/v3/DisplayUploadAdInfo
+        // Creates a new media bundle asset and returns the resource name.
+        $adAssetResourceName = self::createMediaBundleAsset($googleAdsClient, $customerId);
+
+        // Creates a new display upload ad and associates it with the specified ad group.
+        self::createDisplayUploadAdGroupAd(
+            $googleAdsClient,
+            $customerId,
+            $adGroupId,
+            $adAssetResourceName
+        );
+    }
+
+    /**
+     * Creates a media bundle from the assets in a zip file. The zip file contains the HTML5
+     * components.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @return string the resource name of the newly uploaded media bundle asset
+     */
+    private static function createMediaBundleAsset(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId
+    ) {
+        // The HTML5 zip file contains all the HTML, CSS, and images needed for the
+        // HTML5 ad. For help on creating an HTML5 zip file, check out Google Web
+        // Designer (https://www.google.com/webdesigner/).
+        $html5Zip = file_get_contents('https://goo.gl/9Y7qI2');
+
+        // Creates the media bundle asset.
+        $asset = new Asset(
+            [
+                'type' => AssetType::MEDIA_BUNDLE,
+                'media_bundle_asset' => new MediaBundleAsset(
+                    [
+                        'data' => new BytesValue(['value' => $html5Zip]),
+                    ]
+                )
+            ]
+        );
+
+        // Creates an asset operation.
+        $assetOperation = new AssetOperation();
+        $assetOperation->setCreate($asset);
+
+        // Issues a mutate request to add the asset.
+        $assetServiceClient = $googleAdsClient->getAssetServiceClient();
+        $response = $assetServiceClient->mutateAssets(
+            $customerId,
+            [$assetOperation]
+        );
+
+        // Prints the resource name of the added media bundle asset.
+        /** @var MutateAssetResult $addedMediaBundleAsset */
+        $addedMediaBundleAsset = $response->getResults()[0];
```

Comment:
nit: you may want to use a variable for the resource name here, something similar to `$adGroupAdResourceName = $adGroupAdResponse->getResults()[0]->getResourceName();`

This is meaningful here because you request twice the `getResourceName()` method